### DISCUSSION TITLE
Sort by relation count?

### DISCUSSION THREAD

**Discussion by Nv7-GitHub** [pocketbase/pocketbase#1615] [Q&A] [answered]
Is it possible to sort by the amount of relations? For example:
I have a table called `follows` (image attached) which shows which user has followed another user
<img width="657" alt="image" src="https://user-images.githubusercontent.com/50369270/212757616-ab59ec87-d1d6-4ac6-8723-89005a5d8ac5.png">
How can I sort the `users` table by the number of rows in `follows` that point to that user? Should I structure my data differently?

**Comment by ganigeorgiev** [comment, marked as answer]
If `following` is a multi-`relation` field (aka. "Max select != 1") you should be able to use the new `:length` field modifier introduced with [v0.11.0+](https://github.com/pocketbase/pocketbase/releases/tag/v0.11.0), aka.:

```js
const result = await pb.collection("example").getList(1, 30, {
    sort: "following:length", // or prefix with minus "-" for desc order
})
```

**Reply by hungcrush** [reply]
@ganigeorgiev Yo're genius 😭

**Reply by Nv7-GitHub** [reply]
I have a separate table tracking follows between users though, I don't have a relation in the `users` table. How would I make this?

**Reply by Nv7-GitHub** [reply]
Oh wait, I think I understand, I will add a `followers` field to all users, thank you!

**Reply by Nv7-GitHub** [reply]
I can't add a `followers` field because it doesn't play well with API rules, so I instead added `following` field to every user. But, how do I sort by followers (the number of users where the following field contains a specific id) @ganigeorgiev?

**Reply by ganigeorgiev** [reply]
@Nv7-GitHub You can't sort by the count of a single relation field. You can sort by `:length` only if your field has "Max select != 1".
If you want to sort by the count of another collection you'll have to [implement a custom endpoint](https://pocketbase.io/docs/router/) and execute your [custom db query](https://pocketbase.io/docs/database/).
Eventually in the future you'll be able to execute custom queries from the Admin UI with the planned "view" collection type and/or raw/computed sql field features but for now these are left after the relation picker improvements and goja js app event hook bindings.

**Reply by Nv7-GitHub** [reply]
Ok, thank you!
